src/name.rs

```diff
@@ -399,12 +399,61 @@ pub(crate) struct NamespaceResolver {
     nesting_level: i32,
 }
 
+/// These constants define the [reserved namespaces] for the xml standard.
+///
+/// The prefix `xml` is by definition bound to the namespace name
+/// `http://www.w3.org/XML/1998/namespace`. It may, but need not, be declared, and must not be
+/// undeclared or bound to any other namespace name. Other prefixes must not be bound to this
+/// namespace name, and it must not be declared as the default namespace.
+///
+/// The prefix `xmlns` is used only to declare namespace bindings and is by definition bound
+/// to the namespace name `http://www.w3.org/2000/xmlns/`. It must not be declared or
+/// undeclared. Other prefixes must not be bound to this namespace name, and it must not be
+///  declared as the default namespace. Element names must not have the prefix `xmlns`.
+///
+/// [reserved namespaces]: https://www.w3.org/TR/xml-names11/#xmlReserved
+const RESERVED_NAMESPACE_XML: (Prefix, Namespace) = (
+    Prefix(b"xml"),
+    Namespace(b"http://www.w3.org/XML/1998/namespace"),
+);
+const RESERVED_NAMESPACE_XMLNS: (Prefix, Namespace) = (
+    Prefix(b"xmlns"),
+    Namespace(b"http://www.w3.org/2000/xmlns/"),
+);
+
+const RESERVED_NAMESPACES: [(Prefix, Namespace); 2] =
+    [RESERVED_NAMESPACE_XML, RESERVED_NAMESPACE_XMLNS];
+
 impl NamespaceResolver {
+    pub fn new_root_resolver_and_buf() -> (Self, Vec<u8>) {
+        let mut bindings = Vec::new();
+        let mut buffer = Vec::new();
+        for ent in RESERVED_NAMESPACES {
+            let prefix = ent.0.into_inner();
+            let uri = ent.1.into_inner();
+            bindings.push(NamespaceEntry {
+                start: buffer.len(),
+                prefix_len: prefix.len(),
+                value_len: uri.len(),
+                level: 0,
+            });
+            buffer.extend(prefix);
+            buffer.extend(uri);
+        }
+
+        (
+            Self {
+                bindings,
+                ..Self::default()
+            },
+            buffer,
+        )
+    }
     /// Begins a new scope and add to it all [namespace bindings] that found in
     /// the specified start element.
     ///
     /// [namespace binding]: https://www.w3.org/TR/xml-names11/#dt-NSDecl
-    pub fn push(&mut self, start: &BytesStart, buffer: &mut Vec<u8>) {
+    pub fn push(&mut self, start: &BytesStart, buffer: &mut Vec<u8>) -> Result<()> {
```

Comment:
This is a breaking change (which is fine) but it should probably have its own callout in the changelog.

Reply:
Does this require a major version bump? Also, should I add a completely separate entry for the return type change, or should I add it to the existing entry?

Reply:
No, `NamespaceResolver` is a private struct, so this is not a breaking change

Reply:
Given that, I'm assuming that this doesn't need to be cashed out in the changelog.

Reply:
yes, apologies.  oversight on my part